src/prefect/core/flow.py

```diff
@@ -325,9 +325,24 @@ def replace(self, old: Task, new: Task, validate: bool = True) -> None:
 
     @contextmanager
     def _flow_context(self) -> Iterator["Flow"]:
-        with prefect.context(flow=self):
+        """
+        When entering a flow context, the Prefect context is modified to include:
+            - `flow`: the flow itself
+            - `_new_task_tracker`: a set of all tasks created while the context is 
+                open, in order to provide user friendly warnings if they aren't added
+                to the flow itself. This is purely for user experience.
+        """
+        new_task_tracker = set()  # type: Set[Task]
+
+        with prefect.context(flow=self, _new_task_tracker=new_task_tracker):
             yield self
 
+        if new_task_tracker.difference(self.tasks):
+            warnings.warn(
+                "Tasks were created but not added to the flow: "
```

Comment:
This is excellent - only question I have is whether we think this message is sufficient, or should we say more?  e.g., link to https://docs.prefect.io/core/advanced_tutorials/task-guide.html#adding-tasks-to-flows and / or saying "This commonly occurs with Parameter tasks that are unused"